acvm/src/lib.rs

```diff
@@ -58,45 +65,45 @@ pub trait PartialWitnessGenerator {
     fn solve(
         &self,
         initial_witness: &mut BTreeMap<Witness, FieldElement>,
-        mut opcodes: Vec<Opcode>,
+        mut gates_to_resolve: Vec<Opcode>,
```

Comment:
Changes in here seem to be unrelated to adding a block opcode, these should be done in a separate PR so that this PR is only concerned with adding the block opcode

Reply:
Yes, @TomAFrench already suggested this. I will remove this part and create another PR.
For the record, I copied the non-recursive solved from a long-standing PR which included it, but I should not have include this refactoring.

Reply:
The non-recursive solver is required for solving the block opcode, so this is related. I could/should have done it in a separate PR but I did not anticipated this, it is only at the end of my implementation that I realised that I needed it. At that time I just wanted to commit the PR. But yes, I understand your point.